transforms/universal/pdf2md/python/src/pdf2md_transform.py

```diff
@@ -0,0 +1,346 @@
+        # self.page_counter = Counter("worker_pdf_pages_count", "Number of PDF pages converted by the worker")
+        # self.page_convert_gauge = Gauge("worker_pdf_page_avg_convert_time", "Average time for converting a single PDF page on each worker")
+        # self.doc_convert_gauge = Gauge("worker_pdf_convert_time", "Time spent converting a single document")
+
+    def _update_metrics(self, num_pages: int, elapse_time: float):
+        pass
+        # self.page_convert_gauge.set(elapse_time / num_pages)
+        # self.doc_convert_gauge.set(elapse_time)
+        # self.doc_counter.inc(1)
+        # self.page_counter.inc(num_pages)
+
+    def _convert_pdf2md(self, doc_filename:str, ext: str, content_bytes: bytes) -> dict:
+        # Convert PDF to Markdown
+        start_time = time.time()
+        buf = io.BytesIO(content_bytes)
+        input_docs = DocumentStream(filename=doc_filename, stream=buf)
+        input = DocumentConversionInput.from_streams([input_docs])
+
+        converted_docs = self._converter.convert(input)
+        doc: ConvertedDocument = next(converted_docs, None)
+        if doc is None or doc.output is None:
+            raise RuntimeError("Failed in converting.")
+        elapse_time = time.time() - start_time
+
+        content_string = doc.render_as_markdown()
+        num_pages = len(doc.pages)
+        num_tables = len(doc.output.tables if doc.output.tables is not None else 0)
+        num_doc_elements = len(doc.output.main_text if doc.output.main_text is not None else 0)
+
+        self._update_metrics(num_pages=num_pages, elapse_time=elapse_time)
+
+        file_data = {
+            "filename": TransformUtils.get_file_basename(doc_filename),
+            "contents": content_string,
+            "num_pages": num_pages,
+            "num_tables": num_tables,
+            "num_doc_elements": num_doc_elements,
+            "document_id": TransformUtils.str_to_hash(content_string),
+            "ext": ext,
+            "hash": TransformUtils.str_to_hash(content_string),
+            "size": len(content_string),
+            "date_acquired": datetime.now().isoformat(),
+            "pdf_convert_time": elapse_time,
+        }
+
+        return file_data
+
+    def transform_binary(
+        self, file_name: str, byte_array: bytes
+    ) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        If file_name is detected as a PDF file, it generates a pyarrow table with a single row
+        containing the document converted in markdown format.
+        If file_name is detected as a ZIP archive, it generates a pyarrow table with a row
+        for each PDF file detected in the archive.
+        """
+
+        data = []
+        success_doc_id = []
+        failed_doc_id = []
+        skipped_doc_id = []
+        number_of_rows = 0
+
+        try:
+            root_kind = filetype.guess(byte_array)
+
```

Comment:
you explicitly have file name. Can you just use it ext?

Reply:
file extensions are notoriously known to be bad indicator of the file type. we have seen many ".pdf" that instead and html error pages or ".php" files that are instead PDF, etc